### PR TITLE
Imagemagick image writer

### PR DESCRIPTION
**Pull request by dcoeurjo**
# PR Description

If ImageMagick dependency enabled, this PR adds a new writer to export images to (colored) PNG, JPG or any file format defined in ImageMagick.

# Checklist

- [x] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [x] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [ ] All continuous integration tests pass (Travis & appveyor)

### CODE REVIEW

**Comment by dcoeurjo** [comment]
can someone check this PR ? Travis is having a problem with imagemagick but I'm not able to reproduce the bug..

**Comment by dcoeurjo** [comment]
help please;)

**Comment by JacquesOlivierLachaud** [comment]
The error with imagemagick seems related to a bad config path (some surfing indicates MAGICK_CONFIGURE_PATH but I have not checked it yet). For the internal g++-4.8 compiler error, I don't know. I had also ImageMagick problems on my Mac, I had to reinstall it (another version) and it worked.
I'll have a look.

**Comment by kerautret** [comment]
I checked also on my updated Mac and also on older version but also not able to reproduce the bug.
On the Debian server same result. I look it if perhaps by trying other magic version

**Comment by kerautret** [comment]
@dcoeurjo  I think I got it ;) ! I PR on your branch.
It follows the @JacquesOlivierLachaud mention with an  MAGICK_CONFIG_PATH and with an add of the file itself (perhaps we could remove later the fix when the package will be fixed).

File: src/DGtal/io/writers/MagickWriter.h [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      int w = (anImage.domain().upperBound()[0] - anImage.domain().lowerBound()[0])+1;
+      int h = (anImage.domain().upperBound()[1] - anImage.domain().lowerBound()[1])+1;
+      Magick::Geometry geom(w,h);
+      Magick::Image image(geom,"white");
+      for(auto point: anImage.domain())
+      {
+        Color c = aFunctor( anImage(point) );
+        Magick::ColorRGB  magickc( c.red()/255., c.green()/255., c.blue()/255.) ;
+        image.pixelColor( point[0] + anImage.domain().lowerBound()[0],
+                         (h-1) - (point[1] + anImage.domain().lowerBound()[1]),
```
Comment (kerautret) [inline src/DGtal/io/writers/MagickWriter.h]: anImage.domain().upperBound()[1]-point[1] ?

File: src/DGtal/io/writers/MagickWriter.h [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    // ----------------------- Standard services ------------------------------
+    typedef TImage Image;
+    typedef typename TImage::Value Value;
+    typedef TFunctor Functor;
+    
+    BOOST_CONCEPT_ASSERT((  concepts::CUnaryFunctor<TFunctor, Value, Color> )) ;
+    BOOST_STATIC_ASSERT( (TImage::Domain::dimension == 2) );
+
+    /** 
+     * Export an Image with Imagemagick.
```
Comment (kerautret) [inline src/DGtal/io/writers/MagickWriter.h]: ImageMagick

File: src/DGtal/io/writers/MagickWriter.h [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+     * 
+     * @param filename name of the output file the suffix is used to select
```
Comment (kerautret) [inline src/DGtal/io/writers/MagickWriter.h]: file the -> file, the ?

**Comment by kerautret** [comment]
Another comment, the integration with the GenericWriter could be nice, I am looking it, let me know if you think that I can PR on your branch or if you prefer another PR.

**Comment by dcoeurjo** [comment]
Not tested yet but the integration within GenericReader is already there

File: src/DGtal/io/writers/MagickWriter.h (hunk 1 above)
Comment (dcoeurjo) [reply src/DGtal/io/writers/MagickWriter.h]: Thanks , I’ll check

**Comment by dcoeurjo** [comment]
Sorry, you meant GenericWriter...

File: src/DGtal/io/writers/MagickWriter.h (hunk 1 above)
Comment (dcoeurjo) [reply src/DGtal/io/writers/MagickWriter.h]: that should be the same.

File: src/DGtal/io/writers/MagickWriter.h (hunk 3 above)
Comment (dcoeurjo) [reply src/DGtal/io/writers/MagickWriter.h]: thanks, fixed

**Comment by dcoeurjo** [comment]
Comments fixed. I'm merging this PR, thanks.

**Comment by kerautret** [comment]
ok fine for me ;)

**Pull request merged**